File: node/src/carrier-address.ts
import { HttpClient } from './http-client';
import type { ShipzyConfig } from './http-client';
import type { ApiResult } from './index';
import type { Address, AddressListResponse } from './address';

// ============ Client ============

export class CarrierAddressClient extends HttpClient {
    async list(params: Record<string, unknown> = {}): Promise<ApiResult<AddressListResponse>> {
        return this.request('/api/v1/carrier/sdk/addresses/list', 'POST', params);
    }

    async create(data: Record<string, unknown>): Promise<ApiResult<Address>> {
        return this.request('/api/v1/carrier/sdk/addresses/create', 'POST', data);
    }

    async update(id: string, data: Record<string, unknown>): Promise<ApiResult<Address>> {
        return this.request(`/api/v1/carrier/sdk/addresses/${id}/update`, 'POST', data);
    }
}

File: rust/src/carrier_address.rs
use reqwest::Method;

use crate::error::Result;
use crate::http_client::HttpClient;
use crate::types::AddressItem;

pub struct CarrierAddressClient {
    pub(crate) inner: HttpClient,
}

impl CarrierAddressClient {
    pub fn new(inner: HttpClient) -> Self {
        Self { inner }
    }

    pub async fn list(&self, body: serde_json::Value) -> Result<crate::types::ListResponse<AddressItem>> {
        self.inner
            .request(
                "/api/v1/carrier/sdk/addresses/list",
                Method::POST,
                Some(body),
            )
            .await
    }

    pub async fn create(&self, body: serde_json::Value) -> Result<AddressItem> {
        self.inner
            .request(
                "/api/v1/carrier/sdk/addresses/create",
                Method::POST,
                Some(body),
            )
            .await
    }

    pub async fn update(&self, id: &str, body: serde_json::Value) -> Result<AddressItem> {
        self.inner
            .request(
                &format!("/api/v1/carrier/sdk/addresses/{}/update", id),
                Method::POST,
                Some(body),
            )
            .await
    }
}
